apps/opencs/view/render/cameracontroller.cpp

```diff
@@ -668,28 +668,47 @@ namespace CSVRender
 
         mInitialized = true;
     }
+    
+    void OrbitCameraController::setConstRoll(bool enabled)
+    {
+        mConstRoll = enabled;
+    }
 
     void OrbitCameraController::rotateHorizontal(double value)
     {
         osg::Vec3d eye, center, up;
         getCamera()->getViewMatrixAsLookAt(eye, center, up);
+        osg::Vec3d absoluteUp = osg::Vec3(0,0,1);
 
-        osg::Quat rotation = osg::Quat(value, up);
+        osg::Quat rotation = osg::Quat(value, mConstRoll ? absoluteUp : up);
         osg::Vec3d oldOffset = eye - mCenter;
         osg::Vec3d newOffset = rotation * oldOffset;
 
+        if (mConstRoll)
+            up = rotation * up;
+
         getCamera()->setViewMatrixAsLookAt(mCenter + newOffset, mCenter, up);
     }
 
     void OrbitCameraController::rotateVertical(double value)
     {
         osg::Vec3d eye, center, up;
         getCamera()->getViewMatrixAsLookAt(eye, center, up);
+        osg::Vec3d absoluteUp = osg::Vec3(0,0,1);
 
         osg::Vec3d forward = center - eye;
-        osg::Quat rotation = osg::Quat(value, up ^ forward);
+
+        osg::Vec3d axis = absoluteUp ^ forward;
```

Comment:
This would cause issues if absouteUp and forward were to ever be equal. Have you considered this edge case?

Reply:
True, let me fix that.